libsolidity/interface/CompilerStack.cpp

```diff
@@ -444,17 +445,32 @@ map<string, unsigned> CompilerStack::sourceIndices() const
 	return indices;
 }
 
-Json::Value const& CompilerStack::interface(string const& _contractName) const
+Json::Value const& CompilerStack::contractABI(string const& _contractName) const
 {
-	return metadata(_contractName, DocumentationType::ABIInterface);
+	return contractABI(contract(_contractName));
 }
 
-Json::Value const& CompilerStack::metadata(string const& _contractName, DocumentationType _type) const
+Json::Value const& CompilerStack::contractABI(Contract const& _contract) const
 {
-	return metadata(contract(_contractName), _type);
+	if (m_stackState < AnalysisSuccessful)
+		BOOST_THROW_EXCEPTION(CompilerError() << errinfo_comment("Parsing was not successful."));
+
+	solAssert(_contract.contract, "");
+	std::unique_ptr<Json::Value const>* abi = &_contract.abi;
```

Comment:
The reason this was so complicated before was that this pointer could point to different unique pointers. Since we only have one unique pointer, all occurences of `abi` below should be directly replaced by `_contract.abi`.